### ISSUE TITLE
#1062 and #1068 causes perf degradation

### ISSUE THREAD

**Issue by zhixinwen** [gocql/gocql#1072]
We put requests in a buffer, and pull the buffer async to process them. The buffer is building up easily after pulling in #1062 and #1068.

The issue may be caused by the fact that these two diffs would make write serialized.

**Comment by zhixinwen** [comment]
Feel free to revert these two diffs if no quick fix can be applied

**Comment by Zariel** [comment]
Have you had chance to profile?

**Comment by Zariel** [comment]
I would have thought writes were already serialised inside the tcpconn. The
channel overhead shouldn't be that concerning. We could try using a bufio
writer to ammortize the syscall cost but would need to ensure it correctly
flushes.

It shouldn't be too hard to write a benchmark that sends frames of
different sizes and have a dummy responder just return void rows like the
testserver which should be able to show us what changed with the pr.

**Comment by zhixinwen** [comment]
Yes, I originally thought this would not hurt perf too much as tcpconn has a spin lock inside already. 
I did not have the chance to do a profiling while data piling up, as it happens randomly.
I will benchmark it when possible, but I am pretty occupied recently.

**Comment by Zariel** [comment]
I'm not going to have access to a computer for a little while, so if you
want to submit a pr reverting those 2 commits go ahead.

**Comment by OleksiienkoMykyta** [comment]
The last message is dated at Mar 6, 2018 and there are no more discussions in. However, there is a PR https://github.com/apache/cassandra-gocql-driver/pull/1122 that reverts a PR https://github.com/apache/cassandra-gocql-driver/pull/1062, which was causing data races. Looks like the PR's author forgot to add closes to this issue.
I think it is not relevant anymore.
